aws-lc-rs/src/encoding.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+//! Serialization formats
+
+use crate::buffer::Buffer;
+use crate::encoding::types::{
+    Curve25519SeedBufferType, EcPrivateKeyBinType, EcPrivateKeyRfc5915DerType,
+    EcPublicKeyX509DerType,
+};
+
+mod types {
+    pub struct EcPrivateKeyBinType {
+        _priv: (),
+    }
+
+    pub struct EcPrivateKeyRfc5915DerType {
+        _priv: (),
+    }
+
+    pub struct EcPublicKeyX509DerType {
+        _priv: (),
+    }
+
+    pub struct Curve25519SeedBufferType {
+        _priv: (),
+    }
+}
+
+/// Trait for types that can be serialized into a DER format.
+pub trait AsDer<T> {
+    /// Serializes into a DER format.
+    ///
+    /// # Errors
+    /// Returns Unspecified if serialization fails.
+    fn as_der(&self) -> Result<T, crate::error::Unspecified>;
+}
+
+/// Trait for values that can be serialized into a big-endian format
+pub trait AsBigEndian<T> {
+    /// Serializes into a big-endian format.
+    ///
+    /// # Errors
+    /// Returns Unspecified if serialization fails.
+    fn as_be_bytes(&self) -> Result<T, crate::error::Unspecified>;
+}
+
+/// Elliptic curve private key data encoded as a big-endian fixed-length integer.
+pub type EcPrivateKeyBin = Buffer<'static, EcPrivateKeyBinType>;
```

Comment:
Should the lifetime parameter still be defined on `EcPrivateKeyBin` to give us flexibility?

Reply:
So far the types defined here are all `'static`. I don't foresee a situation where we would need to provide serialization in a form that required a lifetime, and if we did, we could define a separate type for that.

Reply:
The only possibility I could see us needing it is if we wanted to have pre-computed encodings (like we did in other instance of the library today). I'd like to move away from that anyways, but I also want to avoid having plethora of types because we were overly aggressive on not exposing lifetime parameters in the API.